src/librbd/cache/rwl/LogEntry.cc

```diff
@@ -163,6 +163,35 @@ std::ostream &operator<<(std::ostream &os,
   return entry.format(os);
 }
 
+void DiscardLogEntry::writeback(librbd::cache::ImageWritebackInterface &image_writeback,
+                                Context *ctx) {
+  image_writeback.aio_discard(ram_entry.image_offset_bytes, ram_entry.write_bytes,
+                              1, ctx);
```

Comment:
Nit: you pass the discard granularity to `C_DiscardRequest` but then don't use it. It should be stored in the log entry.

Reply:
Done